src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Import/ImportOrchestratorJobTests.cs

```diff
@@ -68,16 +86,561 @@ public async Task GivenAnOrchestratorJobAndWrongEtag_WhenOrchestratorJobStart_Th
+                Arg.Any<ImportJobMetricsNotification>(),
+                Arg.Any<CancellationToken>());
+        }
+
+        [Fact]
+        public async Task GivenAnOrchestratorJob_WhenLastSubJobFailed_ThenImportProcessingExceptionShouldBeThrowAndWaitForOtherSubJobsCompleted()
+        {
+            IImportOrchestratorJobDataStoreOperation fhirDataBulkImportOperation = Substitute.For<IImportOrchestratorJobDataStoreOperation>();
+            RequestContextAccessor<IFhirRequestContext> contextAccessor = Substitute.For<RequestContextAccessor<IFhirRequestContext>>();
+            ILoggerFactory loggerFactory = new NullLoggerFactory();
+            IIntegrationDataStoreClient integrationDataStoreClient = Substitute.For<IIntegrationDataStoreClient>();
+            ISequenceIdGenerator<long> sequenceIdGenerator = Substitute.For<ISequenceIdGenerator<long>>();
+            IMediator mediator = Substitute.For<IMediator>();
+            ImportOrchestratorJobInputData importOrchestratorInputData = new ImportOrchestratorJobInputData();
+            ImportOrchestratorJobResult importOrchestratorJobResult = new ImportOrchestratorJobResult();
+            TestQueueClient testQueueClient = new TestQueueClient();
+            int getJobByGroupIdCalledTime = 0;
+
+            testQueueClient.GetJobByIdFunc = (queueClient, id, _) =>
+            {
+                JobInfo jobInfo = queueClient.JobInfos.First(t => t.Id == id);
+                if (jobInfo.Id == 3)
+                {
+                    jobInfo.Status = JobStatus.Failed;
+                    jobInfo.Result = JsonConvert.SerializeObject(new ImportProcessingJobErrorResult() { Message = "Job Failed"});
+                }
+
+                return jobInfo;
+            };
+            testQueueClient.GetJobByGroupIdFunc = (queueClient, groupId, _) =>
+            {
+                IEnumerable<JobInfo> jobInfos = queueClient.JobInfos.Where(t => t.GroupId == groupId);
+                if (++getJobByGroupIdCalledTime > 1)
+                {
+                    foreach (JobInfo jobInfo in jobInfos)
+                    {
+                        if (jobInfo.Status == JobStatus.Running)
+                        {
+                            jobInfo.Status = JobStatus.Completed;
+                        }
+                    }
+                }
+
+                return jobInfos;
+            };
+            importOrchestratorInputData.CreateTime = Clock.UtcNow;
+            importOrchestratorInputData.BaseUri = new Uri("http://dummy");
+            var inputs = new List<InputResource>();
+            for (int i = 0; i < 3; i++)
+            {
+                string location = $"http://dummy/{i}";
+                inputs.Add(new InputResource() { Type = "Resource", Url = new Uri(location) });
+                ImportProcessingJobInputData processingInput = new ImportProcessingJobInputData()
+                {
+                    ResourceLocation = "http:test",
```

Comment:
>http:test [](http://example.com/codeflow?start=40&length=9)

typo?

Reply:
Sorry, I forgot the "//", I will modify it in next commit.